scripts/setup/restore-backup

```diff
@@ -87,17 +87,27 @@ def restore_backup(tarball_file):
         db_dir = os.path.join(tmp, "zulip-backup", "database")
         os.setresuid(0, 0, 0)
         run(["chown", "-R", POSTGRES_USER, "--", tmp])
-        run(
-            [
-                os.path.join(
-                    settings.DEPLOY_ROOT, "scripts", "setup", "terminate-psql-sessions"
-                ),
-                "zulip",
-                "zulip",
-                "zulip_base",
-            ]
-        )
-        as_postgres = ["su", "-s", "/usr/bin/env", "-", "--", POSTGRES_USER]
+
+        as_postgres = [
+            "su", "-", POSTGRES_USER, "-s", "/usr/bin/env", "-",
+            "PGDATABASE={}".format(settings.DATABASES["default"]["NAME"]),
+            "PGHOST={}".format(settings.DATABASES["default"]["HOST"]),
+            "PGPORT={}".format(settings.DATABASES["default"]["PORT"]),
+            "PGUSER={}".format(settings.DATABASES["default"]["USER"]),
+            "PGPASSWORD={}".format(settings.DATABASES["default"]["PASSWORD"]),
```

Comment:
@timabbott Updated. One issue with this approach is the password gets printed out when the command is run. Is that fine?

Reply:
So, on systems with no postgres password (or no PGHOST), I'm not sure this will work correctly `PGPASSWORD=` may be treated as the empty string, which is different from not passing it; I think we'll need to add individual lines inside a check for whether they exist.

I'm not thrilled with the password being printed to the shell (or even visible via `ps` in the command line, if we were to use something more quiet than `run()`), but I'm not sure I know of a reasonable alternative.

@andersk do you have ideas?

Reply:
`run` accepts all the keyword arguments of `subprocess.Popen`, one of which is `env={…}`, letting us set environment variables without putting them on the command line. We could pass them through with `su -p` (`--preserve-environment`); or since we’re already flipping our own UID between `root` and `zulip` with all the `os.setresuid` calls, we could do more of that instead of using `su`.

Reply:
Hmm, yeah, that makes sense.  @andersk do you want to take over finishing this PR?  It'd be really nice to be able to proceed with deleting the legacy `docker-zulip` backup code.